Pipfile

```diff
@@ -4,11 +4,17 @@ verify_ssl = true
 name = "pypi"
 
 [packages]
-alembic = "==0.9.8"
-"psycopg2" = "==2.7.4"
-SQLAlchemy = "==1.2.5"
+alembic = "*"
```

Comment:
Is there a convention to lock to semver major versions? `npm` has `^` which results in minor releases being upgradeable when installing while not installing breaking major versions. If there is, we should use it instead.

Reply:
afaik, it's `~=` for compatible releases, or something like `1.1.*`.

`pipenv` complained about putting detailed version requirements in the `Pipfile`. as i understand it, `Pipfile`is used to define dependencies and their source in a general sense, whereas concrete versions are managed in `Pipfile.lock`. as long as a the lock file is not re-generated or updated, all specified versions will remain the same. listing more explicit versions in the `Pipfile` would restrict updating the dependencies, but only *if* we decide to update them.

this makes sense to newbie me, but if you say it's necessary or safer to put some requirements in the `Pipfile` we can of course do that.

Reply:
Yes, a lock file should prevent that. Coming from an unstable package system where `^` is pretty much standard, I find that it's a good practice to indicate the major release version the code is depending upon. But this may be different for python ecosystem, so let's skip it.

Reply:
Since major semver release mean _breaking changes_, a `*` dependency is actually pretty meaningless since it is not really true in the first place.